### PR TITLE
Add README for image classification example

### PR DESCRIPTION
**Pull request by yeandy**
Instructions on how to run a Pytorch RunInference example.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Add a link to the appropriate issue in your description, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by yeandy** [comment]
R: @ryanthompson591 @AnandInguva

File: sdks/python/apache_beam/examples/inference/README.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+# Example RunInference API Pipelines
+
+This module contains example pipelines that use the Beam RunInference
+API. <!---TODO: Add link to full documentation on Beam website when it's published.-->
+
+## Pre-requisites
+
+You must have `apache-beam>=2.40.0` installed in order to run these pipelines,
+because the `apache_beam.examples.inference` module was added in that release.
+Using the RunInference API also `torch` to be installed.
```
Comment (ryanthompson591) [inline sdks/python/apache_beam/examples/inference/README.md:29]: This sentence needs to be fixed.

File: sdks/python/apache_beam/examples/inference/README.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+This module contains example pipelines that use the Beam RunInference
+API. <!---TODO: Add link to full documentation on Beam website when it's published.-->
+
+## Pre-requisites
+
+You must have `apache-beam>=2.40.0` installed in order to run these pipelines,
+because the `apache_beam.examples.inference` module was added in that release.
+Using the RunInference API also `torch` to be installed. 
+
+To install for a local pipeline, run:
```
Comment (ryanthompson591) [inline sdks/python/apache_beam/examples/inference/README.md:31]: Not all users will be using pytorch. Maybe change this sentence to:
To use pytorch, first install pytorch:

File: sdks/python/apache_beam/examples/inference/README.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+```
+pip install apache-beam torch==1.11.0
+```
+
+To install for a Dataflow pipeline, refer to these
+[instructions](https://beam.apache.org/documentation/sdks/python-pipeline-dependencies/#pypi-dependencies).
+You'll need to add `torch` to a `requirements.txt` file, and then run your
```
Comment (ryanthompson591) [inline sdks/python/apache_beam/examples/inference/README.md:38]: Maybe move this up to a torch dep section.

File: sdks/python/apache_beam/examples/inference/README.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+on [ImageNet dataset](https://www.image-net.org/) using the MobileNetV2
+architecture.
+
+The pipeline reads the images, performs basic preprocessing, passes them to the
+PyTorch implementation of RunInference, and then writes the predictions
+to a text file in GCS.
+
+### Data
+Data related to RunInference has been staged in
+`gs://apache-beam-ml/` for use with these example pipelines:
```
Comment (ryanthompson591) [inline sdks/python/apache_beam/examples/inference/README.md:65]: I don't know gs://apache-beam-ml will really work as a link or is right.

maybe "staged in apache-beam-testing" will work

Feel free to keep it this way if I'm just wrong or misunderstanding.

Maybe this will link right if the users cloud account is set to apache-beam-testing.

File: sdks/python/apache_beam/examples/inference/README.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+<!---
+Add once benchmark test is released
```
Comment (ryanthompson591) [inline sdks/python/apache_beam/examples/inference/README.md:68]: Is this a todo? Maybe add todo here if so.

File: sdks/python/apache_beam/examples/inference/README.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+  --output predictions.csv \
+  --model_state_dict_path gs://apache-beam-ml/models/imagenet_classification_mobilenet_v2.pt
+```
+
+This will write the output to the `predictions.csv` with contents like:
+```
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005002.JPEG,333
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005003.JPEG,711
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005004.JPEG,286
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005005.JPEG,433
```
Comment (ryanthompson591) [inline sdks/python/apache_beam/examples/inference/README.md:104]: 3 or 4 lines should be adequate.

File: sdks/python/apache_beam/examples/inference/README.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005006.JPEG,290
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005007.JPEG,890
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005008.JPEG,592
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005009.JPEG,406
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005010.JPEG,996
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005011.JPEG,327
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005012.JPEG,573
+```
+where the second item in each line is the integer representing the predicted class of the
+image.
```
Comment (ryanthompson591) [inline sdks/python/apache_beam/examples/inference/README.md:114]: it would be cool if one of the ptransforms in the example joined to integer prediction to the actual name of the image.

for example:
gs://apache-beam-ml/datasets/.....5102.jpeg, horse
gs://apache-beam-ml/datasets/.....5102.jpeg, cheese

etc.

But that is outside of the scope of this PR.

**Review by ryanthompson591 — APPROVED** [review]
Looks good, just a couple small comments.

File: sdks/python/apache_beam/examples/inference/README.md (hunk 7 above)
Comment (yeandy) [reply sdks/python/apache_beam/examples/inference/README.md:114]: I have that for a different example https://github.com/apache/beam/pull/21766

File: sdks/python/apache_beam/examples/inference/README.md (hunk 6 above)
Comment (yeandy) [reply sdks/python/apache_beam/examples/inference/README.md:104]: Fixed.

File: sdks/python/apache_beam/examples/inference/README.md (hunk 5 above)
Comment (yeandy) [reply sdks/python/apache_beam/examples/inference/README.md:68]: Yup, fixed

File: sdks/python/apache_beam/examples/inference/README.md (hunk 1 above)
Comment (yeandy) [reply sdks/python/apache_beam/examples/inference/README.md:29]: Fixed.

**Comment by yeandy** [comment]
I have another PR https://github.com/apache/beam/pull/21766, and once that gets merged I can add instructions for it in the README.md as well.

File: sdks/python/apache_beam/examples/inference/README.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+TODO: Add link to full documentation on Beam website when it's published.
+
+i.e. "See the
+[documentation](https://beam.apache.org/documentation/dsls/dataframes/overview/#pre-requisites)
+for details."
+-->
+
+### Datasets and Models for RunInference
+Data related to RunInference has been staged in
+`gs://apache-beam-ml/` for use with these example pipelines. You can see this by using the [gsutil tool](https://cloud.google.com/storage/docs/gsutil#gettingstarted).
```
Comment (TheNeuralBit) [inline sdks/python/apache_beam/examples/inference/README.md:52]: (optional) maybe link to the cloud console here: https://pantheon.corp.google.com/storage/browser/apache-beam-ml

File: sdks/python/apache_beam/examples/inference/README.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+<!---
+TODO: Add once benchmark test is released
+- `gs://apache-beam-ml/testing/inputs/imagenet_validation_inputs.txt`:
+  text file containing the GCS paths of the images of all 5000 imagenet validation data
+    - gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00000001.JPEG
+    - ...
+    - gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00050000.JPEG
+-->
+- `gs://apache-beam-ml/testing/inputs/it_imagenet_validation_inputs.txt/`:
```
Comment (TheNeuralBit) [inline sdks/python/apache_beam/examples/inference/README.md:79]:
```suggestion
- `gs://apache-beam-ml/testing/inputs/it_imagenet_validation_inputs.txt`:
```

File: sdks/python/apache_beam/examples/inference/README.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+  text file containing the GCS paths of the images of a subset of 15 imagenet
+  validation data
+    - gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00000001.JPEG
+    - ...
+    - gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00000015.JPEG
```
Comment (TheNeuralBit) [inline sdks/python/apache_beam/examples/inference/README.md:84]: (optional) It might be nice to clarify that these sub-bullets are the file contents with something like:
```sh
$ gsutil cat gs://apache-beam-ml/testing/inputs/it_imagenet_validation_inputs.txt
gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00000001.JPEG
...
gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00000015.JPEG
```

**Review by TheNeuralBit — APPROVED** [review]
LGTM, a couple minor suggestions

File: sdks/python/apache_beam/examples/inference/README.md (hunk 8 above)
Comment (yeandy) [reply sdks/python/apache_beam/examples/inference/README.md:52]: Is that link accessible to only Google employees though? would `https://console.cloud.google.com/` be better?

Comment (TheNeuralBit) [reply sdks/python/apache_beam/examples/inference/README.md:52]: whoops, yes it would. I copied the wrong thing :)

Comment (TheNeuralBit) [reply sdks/python/apache_beam/examples/inference/README.md:52]: https://console.cloud.google.com/storage/browser/apache-beam-ml

File: sdks/python/apache_beam/examples/inference/README.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+```
+
+### Pytorch dependencies
+The RunInference API has support for the Pytorch framework. To use Pytorch locally, first install `torch`.
+```
+pip install torch==1.11.0
+```
+
+For installation of the `torch` dependency for Dataflow pipelines, refer to these
+[instructions](https://beam.apache.org/documentation/sdks/python-pipeline-dependencies/#pypi-dependencies).
```
Comment (TheNeuralBit) [inline sdks/python/apache_beam/examples/inference/README.md:40]:
```suggestion
For installation of the `torch` dependency on a distributed runner, like Dataflow, refer to these
[instructions](https://beam.apache.org/documentation/sdks/python-pipeline-dependencies/#pypi-dependencies).
```

**Review by TheNeuralBit — CHANGES_REQUESTED** [review]
Sorry, one more thing I forgot to include in the last review.

File: sdks/python/apache_beam/examples/inference/README.md (hunk 11 above)
Comment (TheNeuralBit) [reply sdks/python/apache_beam/examples/inference/README.md]: (Doesn't need to be exactly that text, just in general Beam docs should mention Dataflow as _a_ distributed runner and be clear that there are others)

Comment (yeandy) [reply sdks/python/apache_beam/examples/inference/README.md]: Thanks, fixed.

File: sdks/python/apache_beam/examples/inference/README.md (hunk 10 above)
Comment (yeandy) [reply sdks/python/apache_beam/examples/inference/README.md]: Added.

File: sdks/python/apache_beam/examples/inference/README.md (hunk 8 above)
Comment (yeandy) [reply sdks/python/apache_beam/examples/inference/README.md]: Done.

**Review by TheNeuralBit — APPROVED** [review]

File: sdks/python/apache_beam/examples/inference/README.md [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+```
+
+For installation of the `torch` dependency on a distributed runner, like Dataflow, refer to these
+[instructions](https://beam.apache.org/documentation/sdks/python-pipeline-dependencies/#pypi-dependencies).
+
+<!---
+TODO: Add link to full documentation on Beam website when it's published.
+
+i.e. "See the
+[documentation](https://beam.apache.org/documentation/dsls/dataframes/overview/#pre-requisites)
```
Comment (tvalentyn) [inline sdks/python/apache_beam/examples/inference/README.md:46]: is this a leftover?

**Review by tvalentyn — APPROVED** [review]

**Comment by TheNeuralBit** [comment]
Merging despite the failed test since this is just a docs change.
